### ISSUE TITLE
[QUESTION] Support for Jakarta namespaces 

### ISSUE THREAD

**Issue by Ortofta** [microsoft/mssql-jdbc#2098]
Are there any plans to release a version of the JDBC driver using the "new" Jakarta namespaces instaed of the old javax equivalents?

**Comment by Jeffery-Wasty** [comment]
Hi @Ortofta,

No current plans, and nothing that I can see in our backlog. I'll bring this to the attention of the rest of the team and let you know if this is something we want to eventually include.

**Comment by Ortofta** [comment]
Thanks, @Jeffery-Wasty.

We are in the process of upgrading our app to Spring 6 / Jakarta EE9 (or 10) - and I assume that your dependencies on for example` javax.transaction` will be a problem.

**Comment by lilgreenbird** [comment]
We have put this into our backlog as a future enhancement. Please let us know if you run into issues when you do your upgrade.

**Comment by dcaillia** [comment]
It looks like i'm stuck on this too (also migrating to spring 6).

java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter

which is now in the jakarta,xml.bind-api:4.0.1 module (the class is now named jakarta.xml.bind.DatatypeConverter).

The JakartaEE9 migration (renaming some javax package names to jakarta) is several years old already. Given there's not much complaints here, i guess people found workarounds that i am not aware of?

**Comment by Ortofta** [comment]
We unfortunately had to pause our upgrade project, but I hope to resume work on it soon<sup>TM</sup> - so I too would appreciate an update on what the plans are to support the Jakarta stuff.

**Comment by David-Engel** [comment]
If someone has already done the research and has full context, can they summarize here what the issue is? Specifically, the answers to these questions, and any other pertinent background info, would help prioritize this ask:

1. What is Jakarta? (I did Google it a bit.)
2. What is the impact of not adding Jakarta support? I.e. What, specifically, does Jakarta support unblock?
3. At a high level, what would the required changes look like in the mssql-jdbc library in order to support Jakarta?
4. What kind of backwards compatibility issues would arise from adding Jakarta support? Or how would backwards compatibility with javax be retained?

**Comment by Ortofta** [comment]
@David-Engel - I have done some research, but I'm not claiming to be an expert in this,

>     1. What is Jakarta? (I did Google it a bit.)

Jakarta EE is the continuation of the Java EE specification. The last Oracle version of Java EE was 8, which was released back in 2017. The Java EE specification was then taken over by the Eclipse Foundation for future development. My understanding is that Oracle will not be developing any new versions of the "old" Java EE. I believe there were some dispute over trademarks, leading to the change of package names from javax.* to jakarta.* from Jakarta EE 9 forwards.

See for example [Wikipedia](https://en.wikipedia.org/wiki/Jakarta_EEl), [Jakarta EE FAQ](https://jakarta.ee/about/faq/)

>     2. What is the impact of not adding Jakarta support? I.e. What, specifically, does Jakarta support unblock?

The impact is that any project using the latest versions of popular frameworks like Spring will not be able to use the SQL Server driver, as they have moved to the Jakarta namespace as of version Spring Framework 6 and Boot 3.0.
[Getting Ready for Spring Framework 6](https://springframework.guru/getting-ready-for-spring-framework-6/)

>     3. At a high level, what would the required changes look like in the mssql-jdbc library in order to support Jakarta?

Identify and replace any dependencies on javax packages and replace them with the corresponding jakarta dependencies.

>     4. What kind of backwards compatibility issues would arise from adding Jakarta support? Or how would backwards compatibility with javax be retained?

My understanding is that you need to maintain two separate versions if you want to support usage of the driver by both legacy javax and modern jakarta applications. It's not a good idea to mix javax and jakarta dependencies.

Please feel free to correct anything that I've gotten wrong!

**Comment by Jeffery-Wasty** [comment]
Hi @Ortofta,

Thank you for the reply. We'll look over this issue again after our release and leave a reply with how we're going to proceed with the issue.

**Comment by jonakeys** [comment]
This is becoming a problem when using RedHat JBoss 8.0 and XA Transactions (Jakarta EE9+). It is end of support soon and 8.1 won't work anymore with the current MSSQL JDBC-driver.

**Comment by rkhalloran** [comment]
Has there been any progress on this matter? We are looking to migrate out to JBoss EAP 8 and Tomcat 10.1 both of which have moved to the Jakarta EE namespace, which has been GA for a few years now.

Failing a useful response from MS, has anyone found a third party driver that supports Jakarta APIs?
